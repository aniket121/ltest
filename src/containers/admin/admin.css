.AdminContainer {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
}

.SubscribersTable {
  width: 80%;
}

.SubscribersTable h2 {
  margin-left: 24px;
}

.SubscriberTableColumns {
  display: flex;
  justify-content: space-between;
  padding: 16px;
  color: rgba(0, 0, 0, 0.84);
  font-size: 84%;
}

.SubscriberTableHeader {
  display: flex;
  justify-content: space-between;
}

.Subscriber {
  display: flex;
  align-items: center;
  flex: 10;
  padding: 8px 12px;
  margin-bottom: 12px;
}

.refreshButton {
  cursor: pointer;
  margin: 20px 24px;
}

.refreshButton:hover {
  font-size: 110%;
}

.actionIcon {
  margin-left: 24px;
  max-width: 24px;
  max-height: 16px;
  cursor: pointer;
}

.emailField {
  flex: 4;
}

.betaField {
  flex: 2;
}

.createdField {
  flex: 2;
}

.actionField {
  flex: 2;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  margin-right: 32px
}

.actionButton {
  background-color: rgb(145, 10, 146);
  padding: 4px 8px;
  color: #fff;
  font-size: 16px;
  border: none;
  border-radius: 2px;
  outline: none;
  cursor: pointer;
}
