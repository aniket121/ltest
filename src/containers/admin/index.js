import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import {
  fetchSubscribers,
  grantBeta,
  deleteSubscriber
} from 'actions/subscribers';
import Paper from 'material-ui/Paper';
import Loader from 'components/Loader';
import trashIcon from 'assets/small_ico_trash@2x.png';
import moment from 'moment';
import styles from './admin.css';

const Action = ({ name, icon, onClick }) =>
  <img className={styles.actionIcon} src={icon} alt={name} onClick={onClick} />;

export const Subscriber = ({ className, item, grantBeta, deleteSubscriber }) =>
  <Paper className={className} zDepth={2}>
    <div className={styles.emailField}>
      {item.email}
    </div>
    <div className={styles.betaField}>
      {item.beta ? item.beta.toString() : 'false'}
    </div>
    <div className={styles.createdField}>
      {moment(item.createdAt).format('MM-DD-YYYY')}
    </div>
    <div className={styles.actionField}>
      <button
        className={styles.actionButton}
        onClick={() => {
          grantBeta(item.subscriberId, !item.beta);
        }}
      >
        {item.beta ? 'Revoke' : 'Grant'}
      </button>
      <Action
        name="delete"
        icon={trashIcon}
        onClick={() => deleteSubscriber(item.subscriberId)}
      />
    </div>
  </Paper>;

export class Admin extends Component {
  componentWillMount = () => {
    if (!this.props.user.admin) {
      this.props.history.push('/destinations');
    }
  };

  componentDidMount = () => {
    this.props.fetchSubscribers();
  };

  handleRefreshClick = () => {
    this.props.fetchSubscribers();
  };

  renderSubscribers() {
    const { subscribers, isLoading } = this.props;

    return (
      <div className={styles.Subscribers}>
        <div className={styles.SubscriberTableColumns}>
          <div className={styles.emailField}>Email</div>
          <div className={styles.betaField}>Beta</div>
          <div className={styles.createdField}>Date Subscribed</div>
          <div className={styles.actionField}>Actions</div>
        </div>
        {isLoading
          ? <div style={{ display: 'flex', justifyContent: 'center' }}>
              <Loader />
            </div>
          : subscribers.map((subscriber, i) =>
              <Subscriber
                key={i}
                className={styles.Subscriber}
                item={subscriber}
                grantBeta={this.props.grantBeta}
                deleteSubscriber={this.props.deleteSubscriber}
              />
            )}
      </div>
    );
  }

  render() {
    return (
      <div className={styles.AdminContainer}>
        <h1>Admin Panel</h1>
        <div className={styles.SubscribersTable}>
          <div className={styles.SubscriberTableHeader}>
            <h2>Subscribers</h2>
            <div
              className={styles.refreshButton}
              onClick={this.handleRefreshClick}
            >
              <i className="material-icons">refresh</i>
            </div>
          </div>
          {this.renderSubscribers()}
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({
  subscribers: { subscribers, isLoading, error }
}) => ({ subscribers, isLoading, error });

export default withRouter(
  connect(mapStateToProps, { fetchSubscribers, grantBeta, deleteSubscriber })(
    Admin
  )
);
