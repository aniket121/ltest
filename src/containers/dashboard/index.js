import React, { Component } from 'react';
import dashboardPlaceholder from 'assets/influencer_stats.jpg';
import styles from './Dashboard.css';

export class Dashboard extends Component {
  render() {
    return (
      <div className={styles.Dashboard}>
        <div className={styles.overlay}>
          <h1>Analytics coming soon</h1>
        </div>
        <img
          src={dashboardPlaceholder}
          alt="influencer_stats"
          className={styles.placeholder}
        />
      </div>
    );
  }
}

export default Dashboard;
