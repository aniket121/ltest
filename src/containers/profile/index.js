import React from 'react';
import Paper from 'material-ui/Paper';
import coverPic from 'assets/bg@x.jpg';
import Avatar from 'material-ui/Avatar';
import avatarPic from 'assets/avatar.jpg';
import { Grid, Row, Col } from 'react-flexbox-grid';
import Divider from 'material-ui/Divider';
import { ListItem } from 'material-ui/List';
import {Card, CardActions, CardHeader, CardMedia, CardTitle, CardText} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import TextField from 'material-ui/TextField';

import ModeEditIcon from 'material-ui/svg-icons/editor/mode-edit';
import FollowersIcon from 'material-ui/svg-icons/action/record-voice-over';
import CreditCardIcon from 'material-ui/svg-icons/action/credit-card';
import CommentIcon from 'material-ui/svg-icons/communication/comment';
import ShareIcon from 'material-ui/svg-icons/social/share';
import LikeIcon from 'material-ui/svg-icons/action/thumb-up';
import SaveIcon from 'material-ui/svg-icons/content/save';
import LiveIcon from 'material-ui/svg-icons/av/play-arrow';
import {purple500, yellow500, blue50} from 'material-ui/styles/colors';

import styles from './profile.css';
import { config } from 'aws-sdk';

class Profile extends React.Component {
  constructor(props) {
    super(props);
    
    this.state = {
      editable: false,
      aboutMe: this.props.user.aboutMe,
      username: this.props.user.name
    };
  }
  
  getJoinDate(){
    var joinDate = new Date(this.props.user.createdAt);
    var joinMonth = joinDate.getMonth()+1;
    return(joinMonth+"/"+joinDate.getDate()+"/"+joinDate.getFullYear())
  };

  renderLetterAvatar() {
    return (
      <div>
      <Avatar
          color={yellow500}
          backgroundColor={purple500}
          size={140}
        >
          {this.props.user.name?this.props.user.name.charAt(0):this.props.user.email.charAt(0)}
        </Avatar>
            
      </div>
    );
  };

  renderProfilePic() {
    return (
      <div >
        <img src={this.props.user.imageUrl} alt={this.props.user.name} className={styles.avatarImage} />
      </div>
    );
  };
  
  renderAboutMe() {
    return(
        <Col xs={12} sm={6} md={4} className={styles.aboutMe}>
          <p className={styles.aboutMe}>{this.props.user.aboutMe}</p>
        </Col>
    );
  };

  changeAboutMe() {
    return(
      <Col xs={12} sm={6} md={4} className={styles.aboutMe}>
         <TextField
            id="text-field-default"
            floatingLabelText="About Me"
            defaultValue={this.state.aboutMe}
            multiLine={true}
            onChange={e => this.setState({ aboutMe: e.target.value })}
        />
      </Col>
    );
  };

  renderUsername(){
    return(
      <div>
        <h3>
          {this.state.username ? this.state.username : 'Your Name'}
        </h3>
      </div>
      );
  };

  changeUsername(){
     return(
      <Col xs={12} sm={6} md={4} className={styles.username}>
         <TextField
            id="text-field-default"
            defaultValue={this.state.username}
            floatingLabelText='Please enter your name'
            hintText='John Doe'
            onChange={e => this.setState({ username: e.target.value })} 
          />
      </Col>
       );
  };

  renderEditButton(){
    return(
      <div>
        <FlatButton   
              icon={<ModeEditIcon />}
              label="Edit"
              labelPosition="before"
              className={styles.editButton}
              style={{minWidth:'auto'}}
              onClick={() => this.setState({ editable: true })}      
              />
      </div>
    );
  };

  renderSaveButton(){
     return(
      <div>
          <FlatButton      
            icon={<SaveIcon />}
            label="Save"
            labelPosition="before"
            color={blue50}
            className={styles.editButton}
            style={{minWidth:'auto'}}
            onClick={() => this.triggerUpdateProfile()}      
            />
      </div>
      );
  };

  triggerUpdateProfile() {
    console.log("profile.triggerUpdateProfile");
    const updateStatus = this.props.updateProfile(this.props.user.email,
                          this.state.username,
                          this.props.userToken,
                          this.props.userTokenProvider,
                          this.state.aboutMe ? this.state.aboutMe : ' ',
                          this.props.user.imageUrl);
    if(updateStatus){
      this.setState({ editable: false });
    }
  };

  render() {
    return (
    <div id='homepage-container'>
      <Paper zDepth={4} className={styles.paper}>
        <Grid fluid>
          <Row center="xs">
            <Col xs={12}>
              {this.state.editable?this.renderSaveButton():this.renderEditButton()}
            </Col>
            <Col xs={12} className={styles.avatar}>
              {this.props.user.imageUrl === ''
                  ? this.renderLetterAvatar()
                  : this.renderProfilePic()}
              {this.state.editable ? this.changeUsername() : this.renderUsername()}
              {this.state.editable ? this.changeAboutMe() : this.renderAboutMe()}
              <p className={styles.joinDate}>
                Member since: {this.getJoinDate()}
              </p>
            </Col>
          </Row>
          <Divider />
          {/* <Row>
            <Col xs={6} sm={4} md={2}>
              <ListItem leftIcon={<FollowersIcon />} >
                654 Followers
              </ListItem>
            </Col>
            <Col xs={6} sm={4} md={2}>
              <ListItem leftIcon={<CreditCardIcon />} >
                $7000 Spent
              </ListItem>
            </Col>
            <Col xs={6} sm={4} md={2}>
              <ListItem leftIcon={<CommentIcon />} >
                465 Comments
              </ListItem>
            </Col>
            <Col xs={6} sm={4} md={2}>
              <ListItem leftIcon={<ShareIcon />} >
                231 Shares
              </ListItem>
            </Col>
            <Col xs={6} sm={4} md={2}>
              <ListItem leftIcon={<LikeIcon />} >
                1654 Likes
              </ListItem>
            </Col>
            <Col xs={6} sm={4} md={2}>
              <ListItem leftIcon={<LiveIcon />} >
                78 Livepins
              </ListItem>
            </Col>
          </Row> */}
        </Grid>
      </Paper>

      {/* <div id='posts'>
        <Grid fluid>
          <h2>Recent Broadcasts</h2>
          <Divider/>
          <Row center='sm' >
            <Col sm={12} md={6} className={styles.postCard}>
              <Card>
                <CardHeader
                  title="URL Avatar"
                  subtitle="Subtitle"
                  avatar={avatarPic}
                />
                <CardMedia
                  overlay={<CardTitle title="Overlay title" subtitle="Overlay subtitle" />}
                >
                  <img src={coverPic} alt="coverPic" />
                </CardMedia>
                <CardText>
                  Description
                </CardText>
                <CardActions>
                  <FlatButton label="Action1" />
                  <FlatButton label="Action2" />
                </CardActions>
              </Card>
            </Col>
            <Col sm={12} md={6} className={styles.postCard}>
              <Card>
                <CardHeader
                  title="URL Avatar"
                  subtitle="Subtitle"
                  avatar={avatarPic}
                />
                <CardMedia
                  overlay={<CardTitle title="Overlay title" subtitle="Overlay subtitle" />}
                >
                  <img src={coverPic} alt="coverPic" />
                </CardMedia>
                <CardText>
                  Description
                </CardText>
                <CardActions>
                  <FlatButton label="Action1" />
                  <FlatButton label="Action2" />
                </CardActions>
              </Card>
            </Col>
          </Row>
        </Grid>
      </div> */}
      
    </div>
    );
  }
}

export default Profile;