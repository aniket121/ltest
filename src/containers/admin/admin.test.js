import React from 'react';
import { Admin } from 'containers/admin';
import { shallow } from 'enzyme';

const createProps = props => ({
  subscribers: [
    { email: 'a@a.com', subscriberId: 'id-a', beta: false },
    { email: 'b@b.com', subscriberId: 'id-b', beta: false },
    { email: 'c@c.com', subscriberId: 'id-c', beta: false }
  ],
  user: {
    admin: false
  },
  history: {
    push: jest.fn()
  },
  ...props
});

describe('<Admin />', () => {
  test('renders snapshot', () => {
    const props = createProps();
    const wrapper = shallow(<Admin {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('redirects if user is not admin', () => {
    const props = createProps();
    const wrapper = shallow(<Admin {...props} />);
    expect(props.history.push).toHaveBeenCalled();
  });

  test('does not redirect if user is admin', () => {
    const props = createProps({ user: { admin: true } });
    const wrapper = shallow(<Admin {...props} />);
    expect(props.history.push).not.toHaveBeenCalled();
  });
});
