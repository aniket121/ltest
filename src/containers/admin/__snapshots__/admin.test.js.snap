exports[`<Admin /> renders snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}>
    <h1>
        Admin Panel
    </h1>
    <div
        className={undefined}>
        <div
            className={undefined}>
            <h2>
                Subscribers
            </h2>
            <div
                className={undefined}
                onClick={[Function]}>
                <i
                    className="material-icons">
                    refresh
                </i>
            </div>
        </div>
        <div
            className={undefined}>
            <div
                className={undefined}>
                <div
                    className={undefined}>
                    Email
                </div>
                <div
                    className={undefined}>
                    Beta
                </div>
                <div
                    className={undefined}>
                    Date Subscribed
                </div>
                <div
                    className={undefined}>
                    Actions
                </div>
            </div>
            <Unknown
                className={undefined}
                deleteSubscriber={undefined}
                grantBeta={undefined}
                item={
                    Object {
                        "beta": false,
                        "email": "a@a.com",
                        "subscriberId": "id-a",
                      }
                } />
            <Unknown
                className={undefined}
                deleteSubscriber={undefined}
                grantBeta={undefined}
                item={
                    Object {
                        "beta": false,
                        "email": "b@b.com",
                        "subscriberId": "id-b",
                      }
                } />
            <Unknown
                className={undefined}
                deleteSubscriber={undefined}
                grantBeta={undefined}
                item={
                    Object {
                        "beta": false,
                        "email": "c@c.com",
                        "subscriberId": "id-c",
                      }
                } />
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className={undefined}>
      <h1>
            Admin Panel
      </h1>
      <div
            className={undefined}>
            <div
                  className={undefined}>
                  <h2>
                        Subscribers
                  </h2>
                  <div
                        className={undefined}
                        onClick={[Function]}>
                        <i
                              className="material-icons">
                              refresh
                        </i>
                  </div>
            </div>
            <div
                  className={undefined}>
                  <div
                        className={undefined}>
                        <div
                              className={undefined}>
                              Email
                        </div>
                        <div
                              className={undefined}>
                              Beta
                        </div>
                        <div
                              className={undefined}>
                              Date Subscribed
                        </div>
                        <div
                              className={undefined}>
                              Actions
                        </div>
                  </div>
                  <Unknown
                        className={undefined}
                        deleteSubscriber={undefined}
                        grantBeta={undefined}
                        item={
                              Object {
                                    "beta": false,
                                    "email": "a@a.com",
                                    "subscriberId": "id-a",
                                  }
                        } />
                  <Unknown
                        className={undefined}
                        deleteSubscriber={undefined}
                        grantBeta={undefined}
                        item={
                              Object {
                                    "beta": false,
                                    "email": "b@b.com",
                                    "subscriberId": "id-b",
                                  }
                        } />
                  <Unknown
                        className={undefined}
                        deleteSubscriber={undefined}
                        grantBeta={undefined}
                        item={
                              Object {
                                    "beta": false,
                                    "email": "c@c.com",
                                    "subscriberId": "id-c",
                                  }
                        } />
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Admin
        history={
                Object {
                        "push": [Function],
                      }
        }
        subscribers={
                Array [
                        Object {
                          "beta": false,
                          "email": "a@a.com",
                          "subscriberId": "id-a",
                        },
                        Object {
                          "beta": false,
                          "email": "b@b.com",
                          "subscriberId": "id-b",
                        },
                        Object {
                          "beta": false,
                          "email": "c@c.com",
                          "subscriberId": "id-c",
                        },
                      ]
        }
        user={
                Object {
                        "admin": false,
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Admin {
        "_reactInternalInstance": [Circular],
        "componentDidMount": [Function],
        "componentWillMount": [Function],
        "context": Object {},
        "handleRefreshClick": [Function],
        "props": Object {
          "history": Object {
            "push": [Function],
          },
          "subscribers": Array [
            Object {
              "beta": false,
              "email": "a@a.com",
              "subscriberId": "id-a",
            },
            Object {
              "beta": false,
              "email": "b@b.com",
              "subscriberId": "id-b",
            },
            Object {
              "beta": false,
              "email": "c@c.com",
              "subscriberId": "id-c",
            },
          ],
          "user": Object {
            "admin": false,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}>
          <h1>
                    Admin Panel
          </h1>
          <div
                    className={undefined}>
                    <div
                              className={undefined}>
                              <h2>
                                        Subscribers
                              </h2>
                              <div
                                        className={undefined}
                                        onClick={[Function]}>
                                        <i
                                                  className="material-icons">
                                                  refresh
                                        </i>
                              </div>
                    </div>
                    <div
                              className={undefined}>
                              <div
                                        className={undefined}>
                                        <div
                                                  className={undefined}>
                                                  Email
                                        </div>
                                        <div
                                                  className={undefined}>
                                                  Beta
                                        </div>
                                        <div
                                                  className={undefined}>
                                                  Date Subscribed
                                        </div>
                                        <div
                                                  className={undefined}>
                                                  Actions
                                        </div>
                              </div>
                              <Unknown
                                        className={undefined}
                                        deleteSubscriber={undefined}
                                        grantBeta={undefined}
                                        item={
                                                  Object {
                                                            "beta": false,
                                                            "email": "a@a.com",
                                                            "subscriberId": "id-a",
                                                          }
                                        } />
                              <Unknown
                                        className={undefined}
                                        deleteSubscriber={undefined}
                                        grantBeta={undefined}
                                        item={
                                                  Object {
                                                            "beta": false,
                                                            "email": "b@b.com",
                                                            "subscriberId": "id-b",
                                                          }
                                        } />
                              <Unknown
                                        className={undefined}
                                        deleteSubscriber={undefined}
                                        grantBeta={undefined}
                                        item={
                                                  Object {
                                                            "beta": false,
                                                            "email": "c@c.com",
                                                            "subscriberId": "id-c",
                                                          }
                                        } />
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}>
          <h1>
                    Admin Panel
          </h1>
          <div
                    className={undefined}>
                    <div
                              className={undefined}>
                              <h2>
                                        Subscribers
                              </h2>
                              <div
                                        className={undefined}
                                        onClick={[Function]}>
                                        <i
                                                  className="material-icons">
                                                  refresh
                                        </i>
                              </div>
                    </div>
                    <div
                              className={undefined}>
                              <div
                                        className={undefined}>
                                        <div
                                                  className={undefined}>
                                                  Email
                                        </div>
                                        <div
                                                  className={undefined}>
                                                  Beta
                                        </div>
                                        <div
                                                  className={undefined}>
                                                  Date Subscribed
                                        </div>
                                        <div
                                                  className={undefined}>
                                                  Actions
                                        </div>
                              </div>
                              <Unknown
                                        className={undefined}
                                        deleteSubscriber={undefined}
                                        grantBeta={undefined}
                                        item={
                                                  Object {
                                                            "beta": false,
                                                            "email": "a@a.com",
                                                            "subscriberId": "id-a",
                                                          }
                                        } />
                              <Unknown
                                        className={undefined}
                                        deleteSubscriber={undefined}
                                        grantBeta={undefined}
                                        item={
                                                  Object {
                                                            "beta": false,
                                                            "email": "b@b.com",
                                                            "subscriberId": "id-b",
                                                          }
                                        } />
                              <Unknown
                                        className={undefined}
                                        deleteSubscriber={undefined}
                                        grantBeta={undefined}
                                        item={
                                                  Object {
                                                            "beta": false,
                                                            "email": "c@c.com",
                                                            "subscriberId": "id-c",
                                                          }
                                        } />
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Admin
    history={
        Object {
            "push": [Function],
          }
    }
    subscribers={
        Array [
            Object {
              "beta": false,
              "email": "a@a.com",
              "subscriberId": "id-a",
            },
            Object {
              "beta": false,
              "email": "b@b.com",
              "subscriberId": "id-b",
            },
            Object {
              "beta": false,
              "email": "c@c.com",
              "subscriberId": "id-c",
            },
          ]
    }
    user={
        Object {
            "admin": false,
          }
    } />,
}
`;
